### PR TITLE
Strict null checks for Terminal, File Editor, and CSV viewer

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: packages/csvviewer/src/widget.ts [hunk 1]
```diff
@@ -112,18 +112,9 @@ class CSVViewer extends Widget implements DocumentRegistry.IReadyWidget {
    * Dispose of the resources used by the widget.
    */
   dispose(): void {
-    if (this._grid === null) {
-      return;
+    if (this._monitor) {
+      this._monitor.dispose();
```
Comment (ian-r-rose) [inline packages/csvviewer/src/widget.ts]: Is there a reason not to follow the `_isDisposed` pattern here?

File: packages/fileeditor-extension/src/index.ts [hunk 2]
```diff
@@ -358,12 +358,16 @@ function activate(app: JupyterLab, restorer: ILayoutRestorer, editorServices: IE
...
+      if (!widget) {
+        return;
+      }
+      let path = widget.context.path;
       return commands.execute('markdownviewer:open', { path });
     },
     isVisible: () => {
       let widget = tracker.currentWidget;
-      return widget && PathExt.extname(widget.context.path) === '.md';
+      return widget && PathExt.extname(widget.context.path) === '.md' || false;
```
Comment (ian-r-rose) [inline packages/fileeditor-extension/src/index.ts]: Is this one necessary? It does not trip on my `tsc`

File: packages/terminal-extension/src/index.ts [hunk 3]
```diff
@@ -188,7 +188,9 @@ function addCommands(app: JupyterLab, services: IServiceManager, tracker: Instan
     execute: args => {
       const name = args['name'] as string;
       // Check for a running terminal with the given name.
-      const widget = tracker.find(value => value.session.name === name);
+      const widget = tracker.find(value => {
+        return value.session && value.session.name === name || false;
```
Comment (ian-r-rose) [inline packages/terminal-extension/src/index.ts]: Same here. Why is the final `false` necessary?

File: packages/csvviewer/src/widget.ts (hunk 1 above)
Comment (blink1073) [inline packages/csvviewer/src/widget.ts]: This was a simple enough pass-through that we didn't need it.

File: packages/fileeditor-extension/src/index.ts (hunk 2 above)
Comment (blink1073) [inline packages/fileeditor-extension/src/index.ts]: Here's what I get:

```
src/index.ts(359,51): error TS2345: Argument of type '{ execute: () => Promise<any> | undefined; isVisible: () => boolean | null; label: string; }' is not assignable to parameter of type 'ICommandOptions'.
  Types of property 'isVisible' are incompatible.
    Type '() => boolean | null' is not assignable to type 'CommandFunc<boolean> | undefined'.
      Type '() => boolean | null' is not assignable to type 'CommandFunc<boolean>'.
        Type 'boolean | null' is not assignable to type 'boolean'.
          Type 'null' is not assignable to type 'boolean'.
```

Comment (ian-r-rose) [inline packages/fileeditor-extension/src/index.ts]: Huh, strange. I guess my C is tripping me up, I didn't really see how `&&` could return anything other than a `boolean`.

**Pull request merged**